code/game/machinery/cryopod.dm

```diff
@@ -187,229 +187,231 @@ var/global/list/frozen_items = list("Alpha"=list(),"Bravo"=list(),"Charlie"=list
-									if(SKILL_SPEC_SCOUT)
-										set_name = "Scout Set"
-									if(SKILL_SPEC_SNIPER)
-										set_name = "Sniper Set"
-
-								if(set_name && !available_specialist_sets.Find(set_name))
-									available_specialist_sets += set_name
-						if("Squad Smartgunner")
-							S.num_smartgun--
-						if("Squad Leader")
-							S.num_leaders--
-					S.count--
-				H.assigned_squad?.clean_marine_from_squad(H,TRUE) //Remove from squad recods, if any.
-
-			ticker.mode.latejoin_tally-- //Cryoing someone out removes someone from the Marines, blocking further larva spawns until accounted for
-
-			//Handle job slot/tater cleanup.
-			if(occupant.mind)
-				RoleAuthority.free_role(RoleAuthority.roles_for_mode[occupant.mind.assigned_role])
-
-				if(occupant.mind.objectives.len)
-					cdel(occupant.mind.objectives)
-					occupant.mind.objectives = null
-					occupant.mind.special_role = null
-
-			//Delete them from datacore.
-			if(PDA_Manifest.len)
-				PDA_Manifest.Cut()
-			for(var/datum/data/record/R in data_core.medical)
-				if((R.fields["name"] == occupant.real_name))
-					data_core.medical -= R
-					cdel(R)
-			for(var/datum/data/record/T in data_core.security)
-				if((T.fields["name"] == occupant.real_name))
-					data_core.security -= T
-					cdel(T)
-			for(var/datum/data/record/G in data_core.general)
-				if((G.fields["name"] == occupant.real_name))
-					data_core.general -= G
-					cdel(G)
-
-			if(orient_right)
-				icon_state = "body_scanner_0-r"
-			else
-				icon_state = "body_scanner_0"
-
+	if(orient_right)
+		icon_state = "body_scanner_0-r"
+	else
+		icon_state = "body_scanner_0"
 
-			occupant.ghostize(0) //We want to make sure they are not kicked to lobby.
-			//TODO: Check objectives/mode, update new targets if this mob is the target, spawn new antags?
 
-			//Make an announcement and log the person entering storage.
-			frozen_crew += "[occupant.real_name]"
+	occupant.ghostize(0) //We want to make sure they are not kicked to lobby.
+	//TODO: Check objectives/mode, update new targets if this mob is the target, spawn new antags?
 
-			announce.autosay("[occupant.real_name] has entered long-term hypersleep storage. Belongings moved to hypersleep inventory.", "Hypersleep Storage System")
-			visible_message("<span class='notice'>[src] hums and hisses as it moves [occupant.real_name] into hypersleep storage.</span>")
+	//Make an announcement and log the person entering storage.
+	frozen_crew += "[occupant.real_name]"
 
-			//Delete the mob.
+	announce.autosay("[occupant.real_name] has entered long-term hypersleep storage. Belongings moved to hypersleep inventory.", "Hypersleep Storage System")
+	visible_message("<span class='notice'>[src] hums and hisses as it moves [occupant.real_name] into hypersleep storage.</span>")
 
-			cdel(occupant)
-			occupant = null
-			stop_processing()
+	//Delete the mob.
+	var/mob/new_player/NP = new()
+	NP.ckey = occupant.ckey
+	if(NP.client) NP.client.change_view(world.view)
```

Comment:
```suggestion
	NP.client?.change_view(world.view)
```

Reply:
Remember not to use single-line control statements in the future, as well.

Reply:
Thanks for the tip!